kubernetes-incubator/cri-o
# [WIP] Use distro. packaging to build cri-o, install and test using simple shell scripts

**- What I did**

Simplify PR integration testing by rewriting the playbook with all imperative actions moved into simple bash scripts, and relying on #924 and ``make`` to handle all the build-details.  This removes the burden of maintaining build details in yet-another-place (former playbook-includes), and provides a single-command interface for both local and automated PR testing.

**- How I did it**

* Remove all details regarding building/installing CRI-O and related components from the integration playbook.
* Re-write the playbook to (as simple as possible) perform an package build,  install, then test.
* Goal is for testing to consist of:
    * Execute some simple smoke-test commands (from  a script).
    * Execute (from script) tests from the integration tests package, if above testing ``exit(0)``.
    * Execute the Kubernetes e2e tests, if above testing ``exit(0)``
* Separately / always, run a artifact-collecting and result-processing playbook - producing required junit-format results.

**- How to verify it**

1. Spin up VMs somewhere/somehow of your choosing.
2. On each VM or on localhost, edit ``inventory`` file.
3. Adding hosts to appropriate groups as indicated by comments, including any required access options.
3. On each VM, or localhost, execute ``./venv-cmd.sh ansible-playbook -i inventory.ini ./main.yml``
4. Observe exit code, examine ``/tmp/artifacts/*`` (VMs) or ``$WORKSPACE/artifacts/*/*`` (localhost).

**- TODO -**
- [x] Make **How to verify it** work for Fedora + ed's spec
- [x] Make  **How to verify it** work for RHEL + make test-rpm
- [x] Make  **How to verify it** work for Fedora + make test-rpm
- [ ] Make  **How to verify it** work for Ubuntu + make test-deb
- [x] Make playbook run integration tests
- [ ] Make playbook run integration tests from built sub-package
- [ ] Make playbook run e2e tests
- [x] Make playbook collect results
- [ ] Make playbook run integration / e2e tests on ubuntu
- [ ] Make playbook parse/format junit for downstream consumption
